x-pack/plugins/ml/common/timefilter.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Subject } from 'rxjs';
+import { timefilter as timefilterDep } from 'ui/timefilter';
+import { RefreshInterval } from 'src/legacy/ui/public/timefilter/timefilter';
+
+interface Time {
+  to: string | undefined;
+  from: string | undefined;
+}
+
+const timefilterUpdate$ = new Subject();
+const refreshIntervalUpdate$ = new Subject();
+
+class MlTimefilter {
+  constructor() {
+    // listen for original timefilter emitted events
+    timefilterDep.on('fetch', this.emitUpdate); // setTime or setRefreshInterval called
+    timefilterDep.on('refreshIntervalUpdate', this.emitRefreshIntervalUpdate); // setRefreshInterval called
+  }
+
+  emitUpdate() {
+    timefilterUpdate$.next();
+  }
+
+  emitRefreshIntervalUpdate() {
+    refreshIntervalUpdate$.next();
+  }
+
+  disableAutoRefreshSelector() {
+    return timefilterDep.disableAutoRefreshSelector();
+  }
+
+  disableTimeRangeSelector() {
+    return timefilterDep.disableTimeRangeSelector();
+  }
+
+  enableAutoRefreshSelector() {
+    return timefilterDep.enableAutoRefreshSelector();
+  }
+
+  enableTimeRangeSelector() {
+    return timefilterDep.enableTimeRangeSelector();
+  }
+
+  isAutoRefreshSelectorEnabled() {
+    return timefilterDep.isAutoRefreshSelectorEnabled;
+  }
+
+  isTimeRangeSelectorEnabled() {
+    return timefilterDep.isAutoRefreshSelectorEnabled;
+  }
+
+  getActiveBounds() {
+    return timefilterDep.getActiveBounds();
+  }
+
+  getRefreshInterval() {
+    return timefilterDep.getRefreshInterval();
+  }
+
+  getTime() {
+    return timefilterDep.getTime();
+  }
+
+  off(event: string) {
+    timefilterDep.off(event, this.emitRefreshIntervalUpdate);
+  }
+
+  // in timefilter dependency - 'fetch', 'refreshIntervalUpdate' emitted
+  setRefreshInterval(interval: RefreshInterval) {
+    timefilterDep.setRefreshInterval(interval);
+  }
+  // in timefilter dependency - 'fetch' is emitted
+  setTime(time: Time) {
+    timefilterDep.setTime(time);
+  }
+  // consumer must call unsubscribe on return value
+  subscribeToUpdates(callback: () => void) {
```

Comment:
The name of this function and `subscribeToRefreshIntervalUpdate` are inconsistent. Maybe just drop the `s` to give `subscribeToUpdate` or maybe `subscribeToTimeFilterUpdate`?

Reply:
`subscribeToUpdates` listens for `fetch` being emitted by the timefilter dependency - this happens both on time update and on refreshInterval update while the `subscribeToRefreshIntervalUpdate` only listens for the refreshInterval update event. That's why I had the naming that way. 

Maybe I can be more explicit? `subscribeToTimeAndRefreshIntervalUpdates`, `subscribeToRefreshIntervalUpdate` and so on. That way it will be very clear which events are being listened to?  Open to ideas on this.

Reply:
Ah ok, in that case I think `subscribeToUpdates` is fine as it is. Just add a comment clarifying that it happens for both types of update event.  Adding a `subscribeToTimeUpdate` probably worth adding for completeness.